core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -239,6 +274,11 @@ public void run() {
         // the aggregator receives actual uses.
         rescheduleIdleTimer();
       }
+      if (graceLoadBalancer != null) {
+        // Exit grace period; timer already rescheduled above.
+        loadBalancer = graceLoadBalancer;
```

Comment:
`loadBalancer` too.  Assuming Atomic reads have similar characteristics to volatile you might want to see if they would be a better fit.